### PR TITLE
README overhaul

### PR DESCRIPTION
**Pull request by psoloway**
1. Remove any "what this affects" sections, except where particularly warranted.
2. Make sure that for each parameter, where applicable, there is a data type and a default value.
3. Make sure that for each parameter that's applicable, there is a note if the parameter is optional.
4. Make sure the links in the README work and are accurate.
5. Update the link in the Contributing section to point here: https://docs.puppetlabs.com/forge/contributing.html
6. General copyediting.
7. Replace "defined type" with "define"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
-#Concat
+#concat
+
+
+[![Build Status](https://travis-ci.org/puppetlabs/puppetlabs-concat.png?branch=master)](https://travis-ci.org/puppetlabs/puppetlabs-concat)
```
Comment (mhaskel) [inline README.md]: We were getting rid of these so we could disable an extra layer of travis-ci testing to ease the backup, so can you remove the build status link?

File: README.md [hunk 2]
```diff
@@ -6,41 +9,40 @@
...
 
-This module constructs files from multiple fragments in an ordered way.
+The concat module lets you construct files from multiple ordered fragments of text.
 
 ##Module Description
 
-This module lets you use many concat::fragment{} resources throughout
-your modules to construct a single file at the end.  It does this through
-a shell (or ruby) script and a temporary holding space for the fragments.
+The concat module lets you gather `concat::fragment` resources from your other modules and order them through a single `concat` resource into a coherent file. It does this through a shell (or Ruby) script and a temporary holding space for the fragments.
```
Comment (mhaskel) [inline README.md]: I think this changed while you were working on the README, but it's just Ruby script now :)

File: README.md [hunk 3]
```diff
@@ -6,41 +9,40 @@
...
 
 ##Setup
 
 ###What concat affects
 
-* Installs concatfragments.[sh|rb] based on platform.
-* Adds a concat/ directory into Puppets `vardir`.
+* Installs `concatfragments.[sh|rb]` according to the platform.
```
Comment (mhaskel) [inline README.md]: just `concatfragments.rb`

File: README.md (hunk 1 above)
Comment (psoloway) [inline README.md]: Will do. Do I read correctly that I should do same in all future README updates?

**Comment by jhoblitt** [comment]
While were at it, could the URL be removed from the repo description?  I'm afraid that the blog posts there discussing the API could mislead someone who doesn't know the history of this module.

**Comment by mhaskel** [comment]
@jhoblitt done

File: README.md [hunk 4]
```diff
@@ -6,41 +6,40 @@
...
-a shell (or ruby) script and a temporary holding space for the fragments.
+The concat module lets you gather `concat::fragment` resources from your other modules and order them through a single `concat` resource into a coherent file. It does this through a Ruby script and a temporary holding space for the fragments.
 
 ##Setup
 
 ###What concat affects
 
-* Installs concatfragments.[sh|rb] based on platform.
-* Adds a concat/ directory into Puppets `vardir`.
+* Installs `concatfragments.rb` according to the platform.
```
Comment (mhaskel) [inline README.md]: I think it should just be 'Installs `concatfragments.rb`'. It used to select which script to install base d on the platform, but now everything just gets the ruby script.

Comment (psoloway) [inline README.md]: Oh sheesh, good catch!

**Comment by jhoblitt** [comment]
\o/

**Pull request merged**